src/components/Account/AccountTransaction/AccountTxPanel.js

```diff
@@ -0,0 +1,143 @@
+    progress: '',
+    setAmount: noop,
+    setReceiver: noop,
+    setAsset: noop,
+    setStep: noop,
+    doTransfer: noop
+  };
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.progress === 'FAILED') {
+      alert('Failed..'); // eslint-disable-line
+    }
+  }
+
+  render() {
+    const { amount, receiver, asset, step } = this.props;
+
+    return (
+      <Panel className={styles.accountPanel} renderHeader={null}>
+        <form className={styles.content}>
+          <h2>Transfer Funds</h2>
+          <div className={styles.inputWrapper}>
+            <Input
+              className={styles.inputs}
+              id="amount"
+              type="number"
+              label="Select Amount"
+              placeholder="0"
+              min="0"
+              step={step}
+              value={amount}
+              disabled={false}
+              onChange={this.handleChangeAmount}
+            />
+            <Select className={styles.inputSelect} value={asset} onChange={this.handleSelect}>
+              {map(assetsForSelect, (item, index) => (
+                <option value={item.value} key={`asset${index}`}>{item.label}</option>
+              ))}
+            </Select>
+            <Input
+              className={styles.inputs}
+              id="recipient"
+              type="text"
+              label="Transfer to"
+              placeholder="AFnWixEza..."
+              value={receiver}
+              disabled={false}
+              onChange={this.handleChangeReceiver}
+            />
+          </div>
+          <hr />
+          <div className={styles.buttonWrapper}>
+            <Button
+              className={classNames(styles.buttons, styles.buttonSecondary)}
+              type="submit"
+              onClick={this.handleTransfer}
+            >
+              <Icon className={styles.Icons} name="add" aria-describedby="add" />
+              Add Account
+            </Button>
+            <Button className={styles.buttons} type="submit" onClick={this.handleTransfer}>
+              <Icon className={styles.Icons} name="transfer" aria-describedby="transfer" />
+              Transfer
+            </Button>
+          </div>
+        </form>
+      </Panel>
+    );
+  }
+
+  handleTransfer = () => {
+    const { doTransfer } = this.props;
+    const { net, asset, amount, receiver, address, wif } = this.props;
+
+    try {
+      doTransfer(net, asset, amount, receiver, address, wif);
+      return null;
+    } catch (e) {
+      return null;
+    }
+  };
+
+  handleSelect = (event) => {
+    const { value } = event.target;
+
+    if (value === NEO) {
+      this.props.setStep('1');
+    }
+    if (value === GAS) {
+      this.props.setStep('0.00000001');
+    }
+
+    this.props.setAsset(event.target.value);
+  };
+
+  handleChangeAmount = (event) => {
+    const { value } = event.target;
+    const finalValue = this.props.asset === NEO ?
+      Number(value).toFixed(0) :
+      Number(value).toFixed(8);
```

Comment:
We'll want to add `bignumber.js` and never use `Number(...)` for converting strings to amounts/values.  We'll end up with floating point issues, and people will either transfer the wrong amounts, or transactions will fail.

Reply:
you do a fine job but visually on video reading was much clearer to newcomers. or use the redirection option